### PR TITLE
e2e tests for noop and wg

### LINKED ISSUE #614: e2e tests for WG
need to add wg to our e2e

### PR DESCRIPTION
**Pull request by vkuznecovas**
Closes #614

### CODE REVIEW

**Comment by vkuznecovas** [comment]
The NATS timeout was a result of the service provider dying. The provider did die due to an out of memory error, which is a result of travis limiting us to 3GB of memory. In order to reduce the memory footprint, the consumer and provider are both now using the lightweight keystore. This allows our e2e suite to fit within 2GB of ram(on my local machine at least).

Thanks @tadovas for pointing me towards #631 which allows for drastic memory footprint reduction.

File: e2e/connection_test.go [hunk 1]
```diff
@@ -150,11 +158,38 @@ func consumerConnectFlow(t *testing.T, tequilapi *tequilapi_client.Client, consu
...
+
+type sessionAsserter func(t *testing.T, session endpoints.SessionDTO)
+
+var serviceTypeAssertionMap = map[string]sessionAsserter{
+	"openvpn":   assertOpenvpn,
+	"noop":      assertNoop,
+	"wireguard": assertWireguard,
+}
+
+func assertOpenvpn(t *testing.T, session endpoints.SessionDTO) {
```
Comment (tadovas) [inline e2e/connection_test.go]: What's the point of different service type assertions ? Function is the same :|

Comment (vkuznecovas) [reply e2e/connection_test.go]: fixed, now we're asserting `assertStatsNotZero` and `assertStatsZero

File: bin/test_e2e [hunk 2]
```diff
@@ -2,6 +2,20 @@
 
 source bin/localnet/functions.sh
 
+if [ -n "$TRAVIS_BUILD_NUMBER" ]
```
Comment (tadovas) [inline bin/test_e2e]: Use something from the list of default travis variables https://docs.travis-ci.com/user/environment-variables/#default-environment-variables

File: bin/test_e2e [hunk 3]
```diff
@@ -2,6 +2,20 @@
 
 source bin/localnet/functions.sh
 
+if [ -n "$TRAVIS" ]
+then
+    echo "Travis detected, installing kernel module"
+    sudo add-apt-repository -y ppa:wireguard/wireguard && \
+    sudo apt-get update && \
+    sudo apt install -y linux-headers-$(uname -r) && \
```
Comment (soffokl) [inline bin/test_e2e]: we can install it in a single line
`sudo apt-get install -y wireguard linux-headers-$(uname -r)`

Comment (soffokl) [reply bin/test_e2e]: And it's better to use single too, not both `apt` and `apt-get`.

File: bin/test_e2e [hunk 4]
```diff
@@ -2,6 +2,20 @@
 
 source bin/localnet/functions.sh
 
+if [ -n "$TRAVIS" ]
+then
+    echo "Travis detected, installing kernel module"
```
Comment (soffokl) [inline bin/test_e2e]:
```suggestion
     echo "Travis detected, installing wireguard kernel module"
```

**Pull request merged**